---
author: Basar Chowdhury
title : Plink examples
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
params:
  filename: "ibd_view.mibs"
---


### Getting started
Just typing plink and specifying a file with no further options is a good 
way to check that the file is intact, and to get some basic summary 
statistics about the file.

```{bash echo=TRUE,results='hide',fig.keep='all'}
## Getting started 
plink --file hapmap1
```

PED and MAP files are plain text files; PED files contain genotype 
information (one person per row) and MAP files contain information on the 
name and position of the markers in the PED file
```{bash echo=FALSE,fig.keep='all'}
cat hapmap1.log
```


### Making a binary PED file

The first thing we will do is to make a binary PED file. This more compact 
representation of the data saves space and speeds up subsequent analysis. 
To make a binary PED file, use the following command.

```{bash echo=TRUE,results='hide',fig.keep='all'}
# Making a binary PED file
plink --file hapmap1 --make-bed --out hapmap1

#  you wanted to create a new file that only includes individuals with high genotyping 
#  (at least 95% complete), you would run:
plink --file hapmap1 --make-bed --mind 0.05 --out highgeno
```


### Working with the binary PED file

```{bash echo=TRUE,error=FALSE,fig.keep='all'}
#Working with the binary PED file
#plink --bfile hapmap1
```
```{bash echo=TRUE,fig.keep='all'}
cat plink.log
```


### Summary statistics: missing rates
Next, we shall generate some simple summary statistics on 
rates of missing data in the file, using the --missing option:


```{bash echo=TRUE,results='hide',fig.keep='all'}
# Summary statistics: missing rates
plink --bfile hapmap1 --missing --out miss_stat
```
Here we see that no individuals were removed for low genotypes 
(MIND > 0.1 implies that we accept people with less than 10 percent missingness).  

The per individual and per SNP (after excluding individuals on the basis of low 
genotyping) rates are then output to the files miss_stat.imiss and miss_stat.lmiss 
respectively. If we had not specified an --out option, the root output filename 
would have defaulted to "plink".
```{bash echo=TRUE,fig.keep='all'}
head miss_stat.imiss
```



### Summary statistics: allele frequencies

Next we perform a similar analysis, except requesting allele frequencies 
instead of genotyping rates. The following command generates a file called 
freq_stat.frq which contains the minor allele frequency and allele codes for 
each SNP.

```{bash echo=TRUE,results='hide',fig.keep='all'}
# To perform a stratified analysis, use the --within option.
plink --bfile hapmap1 --freq --within pop.phe --out freq_stat
```

```{bash echo=TRUE,fig.keep='all'}
head miss_stat.lmiss
```


If we were just interested in a specific SNP, and wanted to know what the 
frequency was in the two populations, you can use the --snp option to select 
this SNP:

```{bash echo=TRUE,results='hide',fig.keep='all'}
plink --bfile hapmap1 --snp rs1891905 --freq --within pop.phe --out snp1_frq_stat

# one could simply use the available command line tools to sort the list of association statistics 
# and print out the top ten 
sort --key=7 -nr as1.assoc | head

# Genotypic and other association models
plink --bfile hapmap1 --model --snp rs2222162 --out mod1
head mod1.model 

## Stratification analysis
plink --bfile hapmap1 --cluster --mc 2 --ppc 0.05 --out str1
head str1.cluster1
```


```{bash echo=TRUE,results='hide',fig.keep='all'}
## Association analysis, accounting for clusters
plink --bfile hapmap1 --mh --within str1.cluster2 --adjust --out aac1
head aac1.cmh.adjusted
plink --bfile hapmap1 --cluster --K 2 --out version3

## Quantitative trait association analysis
plink --bfile hapmap1 --assoc --pheno qt.phe --out quant1

## We request clustered permutation as follows, using the original pairing approach to matching:
plink --bfile hapmap1 --assoc --pheno qt.phe --perm --within str1.cluster2 --out quant2

## Extracting a SNP of interest
plink --bfile hapmap1 --snp rs2222162 --recodeAD --out rec_snp1
```





## Association analysis, accounting for clusters

```{r}
m <- as.matrix(read.table(params$filename))
mds <- cmdscale(as.dist(1-m))
k <- c( rep("green",45) , rep("blue",44) )
plot(mds,pch=20,col=k)
```

